lib/ansible/utils/vars.py

```diff
@@ -88,6 +95,19 @@ def combine_vars(a, b):
         _validate_mutable_mappings(a, b)
         result = a.copy()
         result.update(b)
+
+        for key in b:
+            if key in a:
+                if type(a[key]) is AnsibleVaultEncryptedUnicode:
+                    try:
+                        if a[key] != b[key]:
+                            display.warning("Duplicate variable '%s' detected! The value will be overwritten!" % key)
+                    except AnsibleVaultError:
```

Comment:
There is an exception thrown when trying to decrypt [vault_encrypted_example4_unknown_password](https://github.com/ansible/ansible/blob/devel/test/integration/targets/vault/roles/test_vault_embedded_ids/vars/main.yml#L188):
```yaml
# We dont have a secret for this vaulttext, but nothing references it
# so nothing should ever try to decrypt it. So this is testing that
# we dont require all vaulted vars to be decrypted.
vault_encrypted_example4_unknown_password: !vault |
          $ANSIBLE_VAULT;1.1;AES256
          64316436303566666563393931613833316533346539373635663031376664366131353264366132
          3637623935356263643639313562366434383234633232660a353636666134353030646539643139
          65376235333932353531356666363434313066366161383532363166653762326533323233623431
          3934393962633637330a356337626634343736313339316365373239663031663938353063326665
          30643339386131663336366531663031383030313936356631613432336338313962
```